packages/tpetra/core/src/Tpetra_Details_scaleBlockDiagonal.hpp

```diff
@@ -66,12 +66,11 @@ namespace Details {
 
 template<class MultiVectorType>        
 void inverseScaleBlockDiagonal(const MultiVectorType & blockDiagonal, MultiVectorType & multiVectorToBeScaled) {
-  using SC             = typename MultiVectorType::scalar_type;
   using LO             = typename MultiVectorType::local_ordinal_type;
   using local_mv_type  = typename MultiVectorType::dual_view_type::t_dev;
   using range_type     = Kokkos::RangePolicy<typename MultiVectorType::node_type::execution_space, LO>;
   using namespace KokkosBatched;
-  SC SC_one = Teuchos::ScalarTraits<SC>::one();
+  typename MultiVectorType::impl_scalar_type SC_one = Teuchos::ScalarTraits<typename MultiVectorType::impl_scalar_type>::one();
```

Comment:
That should be fine, but prefer this:
```c++
using IST = typename MultiVectorType::impl_scalar_type;
const IST SC_one = Kokkos::ArithTraits<IST>::one ();
```
You should also just be able to assign 1.0 to IST.